### DISCUSSION TITLE
Docker control for induvial dockers in a stack

### DISCUSSION THREAD

**Discussion by arifroni** [louislam/dockge#123] [Feature Request]
hello, would be nice to have controls (start, stop, log) for individual dockers in a stack. something like below picture.
![image](https://github.com/louislam/dockge/assets/59896374/68e00ee5-2d8b-4d48-97d3-2df334b6257d)

**Comment by Hutch79** [comment]
Would love to see this, since it's very hard to debug something if every container in a stack spams to the same console.

**Comment by samphonic** [comment]
+1
Occasionally I need to make a configuration change or restart one container in a stack, but don't want to interrupt the whole stack.

**Comment by Mattssn** [comment]
+1 Having the option to control individual containers in a stack would make this perfect.

**Comment by cbuxton5200** [comment]
Agree! This would be an awesome addition, similar to "docker compose restart/start/stop myservice"

**Comment by notg10** [comment]
agreed, this would be a great feature

**Comment by mizady** [comment]
I have added start. stop, and restart buttons for the container component to allow for individual containers in a stack to be controlled individually.  I got tired of having to restart an entire stack for one container due to a small configuration change in the container.  I didn't want to start a new discussion but i'm happy to open a pull request if its within the scope of dockge for individual container control. also pushed a docker image with these changes. thejericko/dockge:latest if anyone wanted to test it or use it.

branch can be found at: [https://github.com/mizady/dockge/tree/container-control-buttons](https://github.com/mizady/dockge/tree/container-control-buttons) for further inspection.

**Reply by wsw70** [reply]
It would be awesome if you pushed the changes as a PR

**Reply by mizady** [reply]
went ahead and did so since requested.  https://github.com/louislam/dockge/pull/649

**Comment by corpsell** [comment]
This is what I am looking for as well.

**Comment by wsw70** [comment]
This is the feature that keeps me from switching from completely Portainer. I will switch anyway because I want to have better control but that feature would be awesome (I am not sure why it is not there since the mechanism to restart/stop/etc the whole stack is already in place)

**Comment by wsw70** [comment]
related: https://github.com/louislam/dockge/discussions/245

**Comment by jparker80** [comment]
Would also like to see this feature added.

**Comment by BenRoe** [comment]
Fundamental feature. Please add it.

**Comment by manozzo** [comment]
I think that is the most important feature at all in the project lifetime.

I have to deploy a Dozzle container to be able to see my logs in a decent way.
The column in the middle of screen I think is useless just to use tags and a bash button.

**Comment by reclaimer5146** [comment]
The individual container control recommendation is nice, but I'm mostly +1 for individual log views. 

I'm trying to get off Portainer and not having individual log views is the only reason I'm not switching yet. Other than that, Dockge is looking great so far!

**Comment by hamphh** [comment]
I added container controls including an individual log view in my fork. Look at #844

**Reply by wsw70** [reply]
This is really what dockge should be - still light but with everything one needs *on a daily basis* Thank you very much.

**Comment by bytevictor** [comment]
I came to write this and see that it is already the most requested topic in features. It would be ideal and very easy to implement. In my case, I sometimes organize several tools of the same type in the same stack to have them grouped by type: see Plex and Jellyfin in multimedia; qbittorrent and transmission. Depending on the context, I want to turn one or the other on/off (for example, I only use Plex because my Samsung TV only supports Plex but not Jellyfin, but I don't want the docker always started for no reason).
